File: python/largest-series-product/largest_series_product.py
def largest_product(series, size):
    if size < 0:
        raise ValueError("span must not be negative")

    if size > len(series):
        raise ValueError("span must be smaller than string length")

    if not series.isdigit():
        raise ValueError("digits input must only contain digits")

    max_product = 0

    for i in range(len(series) - size + 1):
        sub_series = series[i:i + size]
        product = 1

        for digit in sub_series:
            product *= int(digit)

        max_product = max(max_product, product)

    return max_product

File: javascript/largest-series-product/largest-series-product.js
//
// This is only a SKELETON file for the 'Largest Series Product' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const largestProduct = (seq,dig) => {
  if(dig>seq.length) throw new Error('Span must be smaller than string length');
  if(/\D/.test(seq)) throw new Error('Digits input must only contain digits');
    if(dig<=0) throw new Error('Span must be greater than zero');
     
  const numbers = []
  for (let i = 0; i <= (seq.length-dig); i++) {
    numbers.push(seq.slice(i,i+dig))
  }
  const products = numbers.map(num=>
    num.split("").reduce((total,d)=>{
    return total*parseInt(d)},1)
  )
  return Math.max(...products)
};